libs/dmlf/include/dmlf/colearn/muddle_learner_networker_impl.hpp

```diff
@@ -0,0 +1,111 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/service_ids.hpp"
+#include "dmlf/colearn/colearn_protocol.hpp"
+#include "dmlf/colearn/update_store.hpp"
+#include "dmlf/networkers/abstract_learner_networker.hpp"
+#include "dmlf/update_interface.hpp"
+#include "logging/logging.hpp"
+#include "muddle/muddle_interface.hpp"
+#include "muddle/rpc/client.hpp"
+#include "muddle/rpc/server.hpp"
+#include "network/service/call_context.hpp"
+#include "oef-base/threading/Taskpool.hpp"
+#include "oef-base/threading/Threadpool.hpp"
+
+namespace fetch {
+namespace dmlf {
+namespace colearn {
+
+class MuddleLearnerNetworkerImpl : public dmlf::AbstractLearnerNetworker
+{
+public:
+  using Taskpool           = oef::base::Taskpool;
+  using Threadpool         = oef::base::Threadpool;
+  using TaskP              = Taskpool::TaskP;
+  using MuddlePtr          = muddle::MuddlePtr;
+  using UpdateInterfacePtr = dmlf::UpdateInterfacePtr;
+  using RpcClient          = fetch::muddle::rpc::Client;
+  using RpcClientPtr       = std::shared_ptr<RpcClient>;
+  using Proto              = ColearnProtocol;
+  using ProtoP             = std::shared_ptr<ColearnProtocol>;
+  using RpcServer          = fetch::muddle::rpc::Server;
+  using RpcServerPtr       = std::shared_ptr<RpcServer>;
+  using Bytes              = byte_array::ByteArray;
+  using Store              = UpdateStore;
+  using StorePtr           = std::shared_ptr<Store>;
```

Comment:
Could this be a unique_ptr? I think it will never be shared

Reply:
i tend to just use shared on the basis that one day it might be. also then I don't have to think about it...